### DISCUSSION TITLE
store and load quotes from an online text or csv file

### DISCUSSION THREAD

**Discussion by kasparsj** [jordwest/news-feed-eradicator#383] [Ideas]
Hi Jordan,

Big thanks, been using the extension for couple of months, and find it extremely useful. Additionally I would like more quotes, or an easier way to add custom ones.

Therefore here are a couple of thoughts what I imagine would be a good solution:
- if anybody could create a text or csv file, publish it somewhere online, and then link News Feed eradicator to that file
- the easiest format would be a text file, with 1 quote on each line. If the script detects "~ author" at the end, then it styles it accordingly, if not then the line is presented as a quote without an author
- there could be multiple builtin files, so the users have a choice already without having to create new files
- everybody could submit a file to github, to share with others. the url could go into a list, that's linked from the extension

What do you think?

**Comment by kolinko** [comment]
I would love that, or at least an option to edit quotes in the settings. Right now, I'm trying to build the plugin locally, so I can finally change the quotes.

**Comment by jordwest** [comment]
This is a cool idea, but fetching remote files is something that I'm not confident to implement in this extension.

Main reason is the security issues - fetching data from a remote url and displaying it inside the context of thousands of logged in Facebook accounts brings with it significant security risks. It's something that would have to be implemented very carefully to avoid any script injections (though WebExtensions helps prevent some of this with sandboxing).

I like the idea of loading/saving quotes from an external file, I'll leave this issue open in case anyone wants to have a go at implementing that

**Comment by lifenautjoe** [comment]
I agree 100% with @jordwest .
The only way I can see this happening is adding some UI for adding custom quotes from a file or manually and storing them on the local storage. 

Edit: Oh wait, I think this is already built in 😅

**Comment by jordwest** [comment]
@achen88 has implemented an import feature, which is in the latest release. I'll leave this issue open though as an export might be on the cards in future.

**Comment by kolinko** [comment]
Thank you for the feature, @achen88 !
